Missing very recent dependency fails build process

Hi!

I bumped into this issue and from my understanding it's not this project build issue per se, but I'd like to find how to fix that.
Build fails at the stage of resolving `jdtls` dependencies, where `jdtls` version used is latest at the moment, `1.30.0.202311171630`. The failure happens (again, from my brief lookup) because this version is in the process of release and at least some of its dependencies exists only as -SNAPSHOT versions (namely the problematic `org.eclipse.search.core`.

Is it possible to bind `jdtls` version for the build config to something that is known to be buildable? And if so, how to do that?

Platform: Linux (also tried on MacOS for the sake of report),
JVM: 
```
java -version       
openjdk version "17.0.9" 2023-10-17
OpenJDK Runtime Environment Temurin-17.0.9+9 (build 17.0.9+9)
OpenJDK 64-Bit Server VM Temurin-17.0.9+9 (build 17.0.9+9, mixed mode, sharing
```

```
git clone ... && cd ...
./mvwn clean install
...
[ERROR] Cannot resolve target definition:
[ERROR]   Software being installed: org.eclipse.jdt.ls.core 1.30.0.202311171630
[ERROR]   Missing requirement: org.eclipse.jdt.ls.core 1.30.0.202311171630 requires 'osgi.bundle; org.eclipse.search.core 3.16.0' but it could not be found
[ERROR] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Java Debug Server for Visual Studio Code :: Parent 0.50.0:
[INFO] 
[INFO] Java Debug Server for Visual Studio Code :: Parent . SUCCESS [  0.103 s]
[INFO] Java Debug Server for Visual Studio Code :: Debugger Core SUCCESS [  7.563 s]
[INFO] Java Debug Server for Visual Studio Code :: Debugger Plugin FAILURE [  7.945 s]
[INFO] Java Debug Server for Visual Studio Code :: P2 Update site SKIPPED
[INFO] Java Debug Server for Visual Studio Code :: Target Platform SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  21.154 s
[INFO] Finished at: 2023-11-17T13:48:36-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to resolve target definition file:/home/gemelen/work/tools/java-debug/com.microsoft.java.debug.target/com.microsoft.java.debug.tp.target: See log for details
```

--- a/com.microsoft.java.debug.target/com.microsoft.java.debug.tp.target
+++ b/com.microsoft.java.debug.target/com.microsoft.java.debug.tp.target
@@ -16,11 +16,8 @@
             <unit id="org.eclipse.equinox.sdk.feature.group" version="0.0.0"/>
             <unit id="org.eclipse.jdt.source.feature.group" version="0.0.0"/>
             <unit id="org.eclipse.sdk.feature.group" version="0.0.0"/>
-            <repository location="https://download.eclipse.org/eclipse/updates/4.29/"/>
-        </location>
-        <location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
             <unit id="org.eclipse.xtext.sdk.feature.group" version="0.0.0"/>
-            <repository location="https://download.eclipse.org/releases/2023-09/"/>
+            <repository location="https://download.eclipse.org/releases/2023-12/202311171000/"/>
         </location>
         <location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
             <unit id="org.jboss.tools.maven.apt.core" version="0.0.0"/>

